src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -529,7 +529,25 @@ public void CloseMethod(BoundStatement body)
 #endif
             )
         {
-            return new SynthesizedLocal(CurrentFunction, TypeWithAnnotations.Create(type), kind, syntax, isPinned, refKind
+            return new SynthesizedLocal(CurrentFunction, TypeWithAnnotations.Create(type), kind, syntax, isPinned, refKind, valEscapeScope: null
+#if DEBUG
+                , createdAtLineNumber, createdAtFilePath
+#endif
+                );
+        }
+
+        public LocalSymbol InterpolatedStringBuilderLocal(
+            TypeSymbol type,
+            uint valEscapeScope,
```

Comment:
Consider just adding valEscapeScope as a default parameter to `SynthesizedLocal`. If the escape is _only_ ever going to be applicable to the the string builder, perhaps we should represent it with a different symbol type.

Reply:
I waffled back and forth on whether to define a separate type or just use `SynthesizedLocal`. I can break out it into a separate type.

---
In reply to: [608906130](https://github.com/dotnet/roslyn/pull/52422#discussion_r608906130) [](ancestors = 608906130)